apptest/vmagent.go

```diff
@@ -156,6 +156,12 @@ func (app *Vmagent) ReloadRelabelConfigs(t *testing.T) {
 	t.Fatalf("relabel configs were not reloaded after SIGHUP signal; previous total: %f, current total: %f", prevTotal, currTotal)
 }
 
+// HTTPAddr returns the address at which the vmagent process is listening
+// for http connections.
+func (app *Vmagent) HTTPAddr() string {
```

Comment:
It would be better to add API specific helper functions to the app. Domain-specific. In this case, `RemotewriteURLRelabelConfig`, `APIStatsRemotewriteURLRelabelConfig`, and then the test should rely on them.